server/src/main/java/org/opensearch/extensions/rest/AuthorizationRequest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.rest;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Request to evaluate user privileges for extension actions
+ *
+ * @opensearch.internal
+ */
+public class AuthorizationRequest extends TransportRequest {
+
+    private String extensionUniqueId;
+    // TODO Switch to principal identifier object that is passed to extension
+    private String requestIssuerIdentity;
+    private String permissionId;
+    private Map<String, Object> params;
+
+    public AuthorizationRequest(String extensionUniqueId, String requestIssuerIdentity, String permissionId, Map<String, Object> params) {
+        this.extensionUniqueId = extensionUniqueId;
+        this.requestIssuerIdentity = requestIssuerIdentity;
```

Comment:
These pieces of information have been added into the request body so the caller could alter they as they saw fit. What about changing the usage so this request is only allowed for the current Subject?

Reply:
@peternied I removed the reference to the requestIssuerIdentity from this class.